File: DB/253_NEW_INTERSMS/NODE.js/sms.js
var http = require('http');

// 修改为您的短信账号
var account="";
// 修改为您的短信密码
var password="";
// 手机号码，格式(区号+手机号码)，例如：8615800000000，其中86为中国的区号
var mobile="8615800000000";
// 修改为您要发送的短信内容
var msg="【253云通讯】您的验证码是123456。如非本人操作，请忽略。";
// var msg="test";
// 短信域名地址
var sms_host = 'intapi.253.com';
// 发送短信地址
var send_sms_uri = '/send/json';
// 查询余额地址
var query_balance_uri = '/balance/json';

send_sms(send_sms_uri,account,password,mobile,msg);

query_blance(query_balance_uri,account,password);

// 发送短信方法
function send_sms(uri,account,password,mobile,msg){
	
    var post_data = { // 这是需要提交的数据 
    'account': account,   
    'password': password, 
    'mobile':mobile,
    'msg':msg,
    };  
    var content =  JSON.stringify(post_data);  
    post(uri,content,sms_host);
	
}
  
// 查询余额方法
function query_blance(uri,content,host){
	
    var post_data = { // 这是需要提交的数据 
    'account': account,   
    'password': password, 
    };  
    var content = JSON.stringify(post_data);  
    post(uri,content,sms_host);
}
  
function post(uri,content,host){
	var options = {  
        hostname: host,
        port: 80,  
        path: uri,  
        method: 'POST',  
        headers: {  
            'Content-Type': 'application/json; charset=UTF-8', 
        }  
    };
    var req = http.request(options, function (res) {  
        console.log('STATUS: ' + res.statusCode);  
        
        res.setEncoding('utf8');  
        res.on('data', function (chunk) {  
            console.log('BODY: ' + chunk);  
        });  
    }); 
   
    req.write(content);  
  
    req.end();   
} 

File: DB/253_NEW_INTERSMS/C(C++)/sms.c
#include <arpa/inet.h>
#include <assert.h>
#include <errno.h>
#include <netinet/in.h>
#include <signal.h>
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <sys/wait.h>
#include <netdb.h>
//#include <accountistd.h>

#define SA struct sockaddr
#define MAXLINE 4096
#define MAXSUB  2000
#define MAXPARAM 2048

#define LISTENQ         1024


//线上线下接口宏开关

#define ONLINE  


extern int h_errno;

int sockfd;



char *hostname = "118.178.16.150";
char *send_sms_uri = "/send/json";
char *query_balance_uri = "/balance/json";



/**
 * * 发http post请求
 * */
ssize_t http_post(char *page, char *poststr)
{
    char sendline[MAXLINE + 1], recvline[MAXLINE + 1];
    ssize_t n;
	snprintf(sendline, MAXSUB,
		"POST %s HTTP/1.1\r\n"
		"Host: intapi.253.com\r\n"
		"Content-type: application/json\r\n"
		"Content-length: %zu\r\n\r\n"
		"%s", page, strlen(poststr), poststr);
		//, page, poststr);
    write(sockfd, sendline, strlen(sendline));
	printf("\n%s", sendline);
	printf("\n--------------------------\n");
    while ((n = read(sockfd, recvline, MAXLINE)) > 0) {
        recvline[n] = '\0';
        printf("%s\n", recvline);
    }
    return n;
}

/**
 * * 查账户余额
 * */
ssize_t get_balance(char *account, char *password)
{
    char params[MAXPARAM + 1];
    char *cp = params;

	sprintf(cp,"{\"account\":\"%s\",\"password\":\"%s\"}", account, password);

    return http_post(query_balance_uri, cp);
}

/**
 * * 发送短信
 * */
ssize_t send_sms(char *account, char *password, char *mobile, char *msg)
{
    char params[MAXPARAM + 1];
    char *cp = params;

	sprintf(cp,"{\"account\":\"%s\",\"password\":\"%s\",\"mobile\":\"%s\",\"msg\":\"%s\"}", account, password, mobile, msg);    

    return http_post(send_sms_uri, cp);
}

int main(void)
{
    struct sockaddr_in servaddr;
    char str[50];

    //建立socket连接
    sockfd = socket(AF_INET, SOCK_STREAM, 0);
    bzero(&servaddr, sizeof(servaddr));
    servaddr.sin_addr.s_addr = inet_addr(hostname);
    servaddr.sin_family = AF_INET;
    servaddr.sin_port = htons(80);
    inet_pton(AF_INET, str, &servaddr.sin_addr);
    connect(sockfd, (SA *) & servaddr, sizeof(servaddr));

	char *account = "";
	char *password = "a.123456987";
	//手机号码，格式(区号+手机号码)，例如：8615800000000，其中86为中国的区号
	char *mobile = "8615800000000";
	//必须带签名
	char *msg = "【253云通讯】您的验证码是123400";

    //get_balance(account, password);
    send_sms(account, password, mobile, msg);
    close(sockfd);
    exit(0);
}


















